examples/credentials.py

```diff
@@ -0,0 +1,28 @@
+import json
+from typing import TypedDict
+
+
+class Credentials(TypedDict):
```

Comment:
There is a also a `Credentials` dataclass defined below. Is this needed ? Can you at least move them in a single file for consistency ?